### ISSUE TITLE
Escaping of square brackets does not work

### ISSUE THREAD

**Issue by jakevdp** [vega/vega-lite#5660]
I have been trying to follow https://vega.github.io/vega-lite/docs/field.html to escape square brackets in field names, and cannot get it to work at all. Here is a representative spec ([vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEDGD2B2AzAlgc0gLjMSA3ZApgO6baRHIAmALgBakAsADEwDRS0Fq3WkDMLAL7tIAWwCGAJwDWpHNViwANtWQAHUvACuSpYOHgolcdXFzI8caIKlIx0wFpxADgCcAVj7RXAJld8ANgAjVyZ3IJ8+F38AnyCAdj5KPnj450gDCDEpWSxINVhkeF5WQ0gCeDhKIvQsUAgsgA9zagBPNRs8gEctcWLkU1VcGxEUAiVKW0aAHWmAOlxxJS0bTIbIVpb2zqgevtVB5GHIUcIJ21bZgG1ERVmAXQzDNcgAEgBnaE4JWx5qNXeGAA9EDhqhxHNUANaFognNkLAgZ9vuJQQRwQ4lAMCKC+HMGHMmHMAFbvBAnMr2cTvAjUd5yQxZKlONyebx+QIhMIRKJuQJxRLJVLpLBXRkNHCNBZLFakACMIlaN0UjywfDWDTIUsWyx2PkVytgqrAco1Esg2plOz4BtuRsYgnF92eIEEQA))
```json
{
  "data": {"values": [
      {"x.value": 1, "y[foo]": 3},
      {"x.value": 2, "y[foo]": 1},
      {"x.value": 3, "y[foo]": 4}
    ]},
  "mark": "point",
  "encoding": {
    "x": {"type": "quantitative", "field": "x\\.value"},
    "y": {"type": "quantitative", "field": "y\\[foo\\]"}
  }
}
```
Escaping of the ``"."`` seems to work as expected, but no amount of backslashes causes the y encoding to work properly.

**Comment by domoritz** [comment]
Looks like we are removing the escaping for field references in scale domains and encode blocks.

**Comment by domoritz** [comment]
I think this is an issue with Vega. We use a utility method to split paths and it removes the escaping.

```
$ vega.splitAccessPath('y\\[foo\\]')
["y[foo]"]
```

I would expect

```
$ vega.splitAccessPath('y\\[foo\\]')
["y\\[foo\\]"]
```

**Comment by domoritz** [comment]
Filed as https://github.com/vega/vega/issues/2287

**Comment by jakevdp** [comment]
Thanks!

**Comment by domoritz** [comment]
Fixed in Vega, test in https://github.com/vega/vega-lite/pull/5730

**Comment by domoritz** [comment]
Actually, looks like I had to make a small change in Vega-Lite as well.

**Comment by kanitw** [comment]
Reopen?

**Comment by domoritz** [comment]
I fixed it in #5730.
